src/main.rs

```diff
@@ -61,27 +63,29 @@ async fn db_connection() -> Client {
 /// Establishes connection to Dapr.
 ///
 /// Adds AppCallbackService which defines pub/sub interaction with Dapr.
-async fn dapr_connection() {
-    let addr = "[::]:50006".parse().unwrap();
-
-    let callback_service = AppCallbackService::default();
+async fn dapr_connection(db_client: Database) {
+    let addr = "[::]:50051".parse().unwrap();
+    let collection: mongodb::Collection<ProductVariant> =
+        db_client.collection::<ProductVariant>("product_variants");
 
-    println!("AppCallback server listening on: {}", addr);
+    let callback_service = AppCallbackService { collection };
 
     // Create a gRPC server with the callback_service.
     TonicServer::builder()
         .add_service(AppCallbackServer::new(callback_service))
         .serve(addr)
         .await
         .unwrap();
+
+    println!("AppCallback server listening on: {}", addr);
```

Comment:
not that familiar with Rust, but is there a real logger in your setup?

Reply:
I have implemented a logger, not sure about Dapr Event though, it looked quite complicated and uses https://docs.rs/tracing/latest/tracing/ instead of logs. Found a few example where they activate tracing in the code, but did not get it to work quickly: https://github.com/search?q=repo%3Ahyperium%2Ftonic+tracing&type=code